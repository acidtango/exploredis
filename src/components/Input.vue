<template>
  <label>
    {{label}}
    <input
    :tabindex="tabindex"
    v-on="listeners"
    :type="type"
    :disabled="disabled"
    :readonly="readonly"
    :value="value"
    :autocomplete="autocomplete"
    :placeholder="placeholder"
    :aria-label="label"
  >
  </label>
</template>

<script>
export default {
  name: "Input",
  props: {
    value: [String, Number],
    disabled: Boolean,
    readonly: Boolean,
    type: {
      type: String,
      default: 'text'
    },
    autocomplete: {
      type: String,
      default: 'off'
    },
    label: String,
    placeholder: String,
    tabindex: String
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: event => this.$emit('input', event.target.value)
      }
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss" scoped>
label {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  color: #6f7788;
  font-weight: bold;
  font-size: 15px;
}
input {
  padding: 10px 12px;
  border-radius: 8px;
  border: 2px solid #cdd9ed;
  color: #cbd1dc;
  font-size: 16px;
  line-height: 0;
  box-sizing: border-box;
  margin: 0.5rem 0 1rem;
}
</style>
