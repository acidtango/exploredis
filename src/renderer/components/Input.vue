<template>
  <label>
    {{label}}
    <input
    v-on="listeners"
    :type="type"
    :disabled="disabled"
    :readonly="readonly"
    :value="value"
    :autocomplete="autocomplete"
    :placeholder="placeholder"
    :aria-label="label"
  >
  </label>
</template>

<script>
export default {
  name: 'Input',
  props: {
    /**
     * Sets the input value
     */
    value: [String, Number],
    /**
     * Sets the input disabled state
     */
    disabled: Boolean,
    /**
     * Sets the input readOnly state
     */
    readonly: Boolean,
    /**
     * Sets the input type
     */
    type: {
      type: String,
      default: 'text',
    },
    /**
     * Sets the input autocomplete state
     */
    autocomplete: {
      type: String,
      default: 'off',
    },
    /**
     * Sets the label text
     */
    label: String,
    /**
     * Sets the placeholder text
     */
    placeholder: String,
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: event => this.$emit('input', event.target.value),
      }
    },
  },
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss" scoped>
label {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  color: #6f7788;
  font-weight: bold;
  font-size: 15px;
}
input {
  padding: 10px 12px;
  border-radius: 8px;
  border: 2px solid #cdd9ed;
  color: #cbd1dc;
  font-size: 16px;
  line-height: 0;
  box-sizing: border-box;
  margin: 0.5rem 0 1rem;
}
</style>

<docs>
This button is amazing, use it responsibly.

## Examples

#### Default:
```jsx
<Input />
```

#### With label:
```jsx
<Input label="email" />
```

#### With placeholder:
```jsx
<Input label="email" placeholder="harry@potter.com" />
```

#### With value:
```jsx
<Input label="email" value="harry@potter.com" />
```

#### Disabled:
```jsx
<Input label="email" value="harry@potter.com" disabled />
```

#### Read only:
```jsx
<Input label="email" value="harry@potter.com" readonly />
```
</docs>
