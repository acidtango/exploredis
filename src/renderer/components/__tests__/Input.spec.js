import { shallowMount } from '@vue/test-utils'
import Input from '@/components/Input.vue'

describe('<Input />', () => {
  it('renders by default', () => {
    const wrapper = shallowMount(Input)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a label', () => {
    const propsData = {
      label: 'Name',
    }
    const wrapper = shallowMount(Input, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a placeholder', () => {
    const propsData = {
      label: 'Name',
      placeholder: 'Name',
    }
    const wrapper = shallowMount(Input, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with disabled state', () => {
    const propsData = {
      label: 'Name',
      placeholder: 'Anakin Skywalker',
      disabled: true,
    }
    const wrapper = shallowMount(Input, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with readOnly state', () => {
    const propsData = {
      label: 'Name',
      placeholder: 'Name',
      readOnly: true,
    }
    const wrapper = shallowMount(Input, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with an initial value', () => {
    const propsData = {
      label: 'Name',
      placeholder: 'Name',
      value: 'General Kenobi',
    }
    const wrapper = shallowMount(Input, { propsData })
    expect(wrapper).toMatchSnapshot()
  })
})
