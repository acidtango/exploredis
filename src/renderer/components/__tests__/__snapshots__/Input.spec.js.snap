// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders a label 1`] = `<label><span>Name</span> <input type="text" autocomplete="off" aria-label="Name"></label>`;

exports[`<Input /> renders a placeholder 1`] = `<label><span>Name</span> <input type="text" autocomplete="off" placeholder="Name" aria-label="Name"></label>`;

exports[`<Input /> renders by default 1`] = `
<label>
  <!----> <input type="text" autocomplete="off"></label>
`;

exports[`<Input /> renders with an initial value 1`] = `<label><span>Name</span> <input type="text" autocomplete="off" placeholder="Name" aria-label="Name"></label>`;

exports[`<Input /> renders with disabled state 1`] = `<label><span>Name</span> <input type="text" disabled="disabled" autocomplete="off" placeholder="Anakin Skywalker" aria-label="Name"></label>`;

exports[`<Input /> renders with readOnly state 1`] = `<label readonly="true"><span>Name</span> <input type="text" autocomplete="off" placeholder="Name" aria-label="Name"></label>`;
